activerecord/lib/active_record/connection_adapters/abstract/transaction.rb

```diff
@@ -205,7 +205,9 @@ def begin_transaction(options = {})
                                        run_commit_callbacks: run_commit_callbacks)
             end
 
-          transaction.materialize! unless @connection.supports_lazy_transactions? && lazy_transactions_enabled?
+          if options[:lazy] == false || !(@connection.supports_lazy_transactions? && lazy_transactions_enabled?)
+            transaction.materialize!
+          end
           @stack.push(transaction)
           @has_unmaterialized_transactions = true if @connection.supports_lazy_transactions?
```

Comment:
Should this condition match the one above? (I see it already doesn't, but I'm not sure whether that's right?)

Reply:
Hum. That's weird indeed. It looks like it should have been part of the conditional but wasn't.

I'll try to figure out what this variable is for.

Reply:
Ok, it's a small optimization flag, and was indeed incorrectly set before.

I fixed it as well.